src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,670 @@
+      var temporaryStyles = [];
+      var classes = element.attr('class');
+      var styles = packageStyles(options);
+      var animationClosed;
+      var animationPaused;
+      var animationCompleted;
+      var deferred;
+
+      if (options.duration === 0 || (!$sniffer.animations && !$sniffer.transitions)) {
+        close();
+        return;
+      }
+
+      var method = options.event && isArray(options.event)
+            ? options.event.join(' ')
+            : options.event;
+
+      // despite any of these methods being a valid event, we special case pure
+      // class-based animations so that we can avoid applying CSS blocking on
+      // the element to allow for normal transitions to work (this is how enter,
+      // leave and move can perform the `ng-EVENT` methods without causing an
+      // unpected transition animation to occur).
+      var structural = method &&
+                        [' setClass ', ' addClass ',' removeClass '].indexOf(' ' + method + ' ') == -1;
+
+      var structuralClassName = '';
+      var addRemoveClassName = '';
+
+      if (structural) {
+        structuralClassName = pendClasses(method, 'ng-', true);
+      }
+
+      if (options.addClass) {
+        addRemoveClassName += pendClasses(options.addClass, '-add');
+      }
+
+      if (options.removeClass) {
+        if (addRemoveClassName.length) {
+          addRemoveClassName += ' ';
+        }
+        addRemoveClassName += pendClasses(options.removeClass, '-remove');
+      }
+
+      var setupClasses = [structuralClassName, addRemoveClassName].join(' ').trim();
+      var fullClassName =  classes + ' ' + setupClasses;
+      var activeClasses = pendClasses(setupClasses, '-active');
+      var hasToStyles = styles.to && Object.keys(styles.to).length > 0;
+
+      // there is no way we can trigger an animation since no styles or
+      // no classes are being applied which would then trigger a transition
+      if (!hasToStyles && !setupClasses) {
+        close();
+        return false;
+      }
+
+      var cacheKey, stagger;
+      if (options.stagger > 0) {
+        var staggerVal = parseFloat(options.stagger);
+        stagger = {
+          transitionDelay: staggerVal,
+          animationDelay: staggerVal,
+          transitionDuration: 0,
+          animationDuration: 0
+        };
+      } else {
+        cacheKey = gcsHashFn(node, fullClassName);
+        stagger = computeCachedCSSStaggerStyles(node, setupClasses, cacheKey, {
+          transitionDelay:    TRANSITION_PROP + DELAY_KEY,
+          transitionDuration: TRANSITION_PROP + DURATION_KEY,
+          animationDelay:     ANIMATION_PROP  + DELAY_KEY,
+          animationDuration:  ANIMATION_PROP  + DURATION_KEY
+        });
+      }
+
+      $$jqLite.addClass(element, setupClasses);
+
+      var applyOnlyDuration;
+
+      if (options.transitionStyle) {
+        var transitionStyle = [TRANSITION_PROP, options.transitionStyle];
+        applyInlineStyle(node, transitionStyle);
```

Comment:
It would seem to me better to just pass in the key and value separately in these cases rather than passing in an array that then needs to be unwrapped inside `applyInlineStyle`. Then we could considerably simplify that function.